feat(i18n): translate src/content/learn/adding-interactivity.md from English to Vietnamese (#471)
reactjs/vi.react.dev: src/content/learn/adding-interactivity.md

---
title: Thêm Tính Tương Tác
---

<Intro>

> Một số thứ trên màn hình cập nhật để phản hồi đầu vào của người dùng. Ví dụ, nhấp vào thư viện ảnh sẽ chuyển đổi ảnh đang hoạt động. Trong React, data thay đổi theo thời gian được gọi là *state*. Bạn có thể thêm state vào bất kỳ component nào và cập nhật nó khi cần thiết. Trong chương này, bạn sẽ học cách viết các component xử lý tương tác, cập nhật state của chúng và hiển thị output khác nhau theo thời gian.

Reviewer: ```suggestion
Một số thứ trên màn hình cập nhật theo phản hồi đầu vào của người dùng. Ví dụ, nhấp vào thư viện ảnh sẽ chuyển đổi ảnh đang được hiện thị. Trong React, data thay đổi theo thời gian được gọi là *state*. Bạn có thể thêm state vào bất kỳ component nào và cập nhật nó khi cần thiết. Trong chương này, bạn sẽ học cách viết các component xử lý tương tác, cập nhật state của chúng và hiển thị output khác nhau theo thời gian.
```

</Intro>

<YouWillLearn isChapter={true}>

* [Cách xử lý các sự kiện do người dùng khởi tạo](/learn/responding-to-events)
* [Cách làm cho component "nhớ" thông tin với state](/learn/state-a-components-memory)
* [Cách React cập nhật UI trong hai giai đoạn](/learn/render-and-commit)
* [Tại sao state không cập nhật ngay sau khi bạn thay đổi nó](/learn/state-as-a-snapshot)

> * [Cách xếp hàng đợi nhiều cập nhật state](/learn/queueing-a-series-of-state-updates)

Reviewer: ```suggestion
* [Cách xử lí nhiều lần cập nhật state liên tiếp](/learn/queueing-a-series-of-state-updates)
```

<LearnMore path="/learn/responding-to-events">

Đọc **[Phản Hồi Sự Kiện](/learn/responding-to-events)** để học cách thêm event handler.

</LearnMore>

## State: bộ nhớ của component {/*state-a-components-memory*/}

> Các component thường cần thay đổi những gì hiển thị trên màn hình kết quả của một tương tác. Gõ vào form sẽ cập nhật trường input, nhấp "tiếp theo" trên carousel ảnh sẽ thay đổi ảnh nào được hiển thị, nhấp "mua" sẽ đưa sản phẩm vào giỏ hàng. Các component cần "nhớ" các thứ: giá trị input hiện tại, ảnh hiện tại, giỏ hàng. Trong React, loại bộ nhớ cụ thể của component này được gọi là *state*.

Reviewer: ```suggestion
Các component thường cần thay đổi những gì hiển thị trên màn hình theo kết quả của một tương tác. Gõ vào form sẽ cập nhật trường input, nhấp "tiếp theo" trên carousel ảnh sẽ thay đổi ảnh nào được hiển thị, nhấp "mua" sẽ đưa sản phẩm vào giỏ hàng. Các component cần "nhớ" các thứ: giá trị input hiện tại, ảnh hiện tại, giỏ hàng. Trong React, loại bộ nhớ cụ thể của component này được gọi là *state*.
```

<LearnMore path="/learn/updating-objects-in-state">

Đọc **[Cập Nhật Object Trong State](/learn/updating-objects-in-state)** để học cách cập nhật object một cách chính xác.

</LearnMore>

## Cập nhật array trong state {/*updating-arrays-in-state*/}

> Array là một loại object JavaScript có thể thay đổi khác mà bạn có thể lưu trữ trong state và nên coi như chỉ đọc. Giống như với object, khi bạn muốn cập nhật một array được lưu trữ trong state, bạn cần tạo một cái mới (hoặc tạo một bản sao của một cái hiện có), và sau đó thiết lập state để sử dụng array mới:

Reviewer: ```suggestion
Array là một loại object JavaScript có thể thay đổi (mutable) khác mà bạn có thể lưu trữ trong state, nhưng nên coi như chỉ đọc được (read-only). Giống như với object, khi bạn muốn cập nhật một array được lưu trữ trong state, bạn cần tạo một cái mới (hoặc tạo một bản sao của một cái hiện có), và sau đó thiết lập state để sử dụng array mới:
```
